sharktank/sharktank/ops/sharded_impls.py

```diff
@@ -204,6 +204,36 @@ def sharded_elementwise_binary(
     return ShardedPrimitiveTensor(shard_dim=x.shard_dim, shape=x.shape, ts=partials)
 
 
+@elementwise.override(ReplicatedTensor, ShardedPrimitiveTensor)
+def elementwise_binary_replicated_lhs_sharder_rhs(
+    operator, x: ReplicatedTensor, y: ShardedPrimitiveTensor
+):
+    assert x.shard_count == y.shard_count
```

Comment:
This will be a user error message: Fine to still assert, but include a more detailed message. For something like this, better to do it in an if/raise.